src/pathfind/teleport.cpp

```diff
@@ -43,19 +45,19 @@ namespace {
 // This constructor is *only* meant for loading from saves
 teleport_group::teleport_group(const config& cfg) : cfg_(cfg), reversed_(cfg["reversed"].to_bool(false)), id_(cfg["id"])
 {
-	assert(cfg.has_attribute("id"));
-	assert(cfg.has_attribute("reversed"));
+	VALIDATE(cfg.has_attribute("id"), missing_mandatory_wml_key("cfg", "id"));
+	VALIDATE(cfg.has_attribute("reversed"), missing_mandatory_wml_key("cfg", "reversed"));
 
-	assert(cfg_.child_count("source") == 1);
-	assert(cfg_.child_count("target") == 1);
-	assert(cfg_.child_count("filter") == 1);
+	VALIDATE(cfg_.child_count("source") == 1, _("config should have one 'source' child."));
+	VALIDATE(cfg_.child_count("target") == 1, _("config should have one 'target' child."));
+	VALIDATE(cfg_.child_count("filter") == 1, _("config should have one 'filter' child."));
```

Comment:
I'd support adding an equivalent of `missing_mandatory_wml_key` which says "In section ‘[$section|]’ the key ‘$key|’ is set more than once." along with a `VALIDATE_EXACTLY_ONE_CHILD` macro which uses `missing_mandatory_wml_key` or the new message as appropriate.

Reply:
It's still better than assert though.